.github/workflows/soroban-rpc.yml

```diff
@@ -25,19 +29,69 @@ jobs:
       - uses: ./.github/actions/setup-go
         with:
           go-version: ${{ matrix.go }}
-      - run: go test -race -cover -timeout 25m -v ./cmd/soroban-rpc/...
+      - run: |
+          make build-libpreflight
+          go test -race -cover -timeout 25m -v ./cmd/soroban-rpc/...
+
+  build:
+    name: Build
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-latest
+            rust_target: x86_64-unknown-linux-gnu
+            go_arch: amd64
+          - os: ubuntu-latest
+            rust_target: aarch64-unknown-linux-gnu
+            go_arch: arm64
+          - os: macos-latest
+            rust_target: x86_64-apple-darwin
+            go_arch: amd64
+          - os: macos-latest
+            rust_target: aarch64-apple-darwin
+            go_arch: arm64
+          - os: windows-latest
+            rust_target: x86_64-pc-windows-gnu
+            go_arch: amd64
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v3
+      # we cannot use our own ./.github/actions/setup-go action
+      # because it uses apt-get and some OSs (e.g. windows) don't have it
+      - uses: actions/setup-go@v3
+        with:
+          go-version: 1.19.1
+
+      - run: rustup update
+      - run: rustup target add ${{ matrix.rust_target }}
+
+      # Use cross-compiler for linux aarch64
+      - if: matrix.rust_target == 'aarch64-unknown-linux-gnu'
+        name: Install aarch64 cross-compilation toolchain
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y gcc-10-aarch64-linux-gnu
+          echo 'CC=aarch64-linux-gnu-gcc-10' >> $GITHUB_ENV
+
+      - name: Build Soroban RPC reproducible build
+        run: |
+          CARGO_BUILD_TARGET=${{ matrix.rust_target }} make build-libpreflight
+          CGO_ENABLED=1 GOARCH=${{ matrix.go_arch }} go build -trimpath -buildvcs=false ./cmd/soroban-rpc
+          touch empty
+          ls -lh soroban-rpc
+          file soroban-rpc
```

Comment:
What's the `touch empty` for?

Reply:
I don't recall, it was there before the PR got started. I will remove it